K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs

```diff
@@ -26,7 +27,15 @@ public void Visit(AggregationContainer aggregationContainer)
 
             if (aggregationContainer.PrimaryAggregation is BucketAggregation bucketAggregation)
             {
-                bucketAggregation.SubAggregationsKustoQL = BuildSummarizableMetricsQuery(aggregationContainer.SubAggregations);
+                VisitedMetrics.Add(EncodeKustoField(bucketAggregation.Key));
```

Comment:
Again, I'm going to suggest to cleanup a bit, tell me what do you think.
It could be even cleaner if ISummarizable and IPartitionable were subclasses of Aggregate rather than interfaces (does it make sense for them to be interfaces on anything else?)

```c#
        public void Visit(AggregationContainer aggregationContainer)
        {
            Ensure.IsNotNull(aggregationContainer, nameof(aggregationContainer));

            if (aggregationContainer.PrimaryAggregation == null)
            {
                return;
            }

            if (aggregationContainer.PrimaryAggregation is BucketAggregation bucketAggregation)
            {
                VisitedMetrics.Add(EncodeKustoField(bucketAggregation.Key));
                VisitedMetrics.Add(EncodeKustoField(bucketAggregation.MetricKey));

                var primaryAggregations = aggregationContainer.SubAggregations.Values.Select(x => x.PrimaryAggregation).ToList();

                bucketAggregation.SummarizableMetricsKustoQL = BuildSummarizableMetricsQuery(primaryAggregations.Where(x => x is ISummarizable));

                bucketAggregation.PartitionableMetricsKustoQL = BuildPartitionableMetricsQuery(primaryAggregations.Where(x => x is IPartitionable), bucketAggregation.Key);
            }

            aggregationContainer.PrimaryAggregation.Accept(this);
            aggregationContainer.KustoQL = aggregationContainer.PrimaryAggregation.KustoQL;
        }

        private string BuildSummarizableMetricsQuery(IEnumerable<Aggregation> aggregations)
        {
            var query = new StringBuilder();

            // Collect all ISummarizable metrics
            // ['2']=max(AvgTicketPrice), ['3']=avg(DistanceKilometers)
            var summarizableMetrics = new List<string>();

            foreach (var aggregation in aggregations)
            {
                aggregation.Accept(this);
                summarizableMetrics.Add($"{aggregation.KustoQL}");
            }

            var summarizableMetricsExpression = string.Join(',', summarizableMetrics);
            query.Append($"{summarizableMetricsExpression}");

            if (!string.IsNullOrEmpty(summarizableMetricsExpression))
            {
                query.Append(',');
            }

            return query.ToString();
        }

        private string BuildPartitionableMetricsQuery(IEnumerable<Aggregation> aggregations, string partitionKey)
        {
            // Collect all additional queries built from IPartitionable metrics
            var query = new StringBuilder();

            foreach (var aggregation in aggregations)
            {
                ((IPartitionable)aggregation).PartitionKey = partitionKey;
                aggregation.Accept(this);
                query.Append($"{aggregation.KustoQL}");
            }

            return query.ToString();
        }
        ```

Reply:
I agree for changing the input arg to IEnumerable<Aggregation> aggregations. But I prefer to apply the "filter" aggregations.Where(x => x is ISummarizable) inside BuildSummarizableMetricsQuery as this method cannot be used for anything else than ISummarizable metrics. Same for partitions.

Reply:
Refactoring done.